pennylane/__init__.py

```diff
@@ -338,3 +307,45 @@ def run_cnot():
 def version():
     """Returns the PennyLane version number."""
     return __version__
+
+
+# add everything as long as it's not a module and not prefixed with _
+_all = sorted(
+    [
+        name
+        for name, function in globals().items()
+        if not (name.startswith("_") or isinstance(function, types.ModuleType))
+    ]
+)
+
+
+_qchem = None
+
+
+def __getattr__(name):
+    """Ensure that the qchem module is imported lazily"""
+    if name == "qchem":
+        global _qchem  # pylint: disable=global-statement
+
+        if _qchem is None:
+
+            for entry in pkg_resources.iter_entry_points("pennylane.qchem"):
+                if entry.name == "OpenFermion":
+                    _qchem = entry.load()
+
+            if _qchem is None:
+                raise ImportError(
```

Comment:
This seems to check if OpenFermion was found as an entry. What if we have PennyLane-QChem installed, but OpenFermion uninstalled (e.g., first Qchem was installed, but then OpenFermion got manually uninstalled)? We'd get the `ImportError`, however, it refers to QChem not Openfermion, correct?

Reply:
oh this is an historical accident 😆 The entry point for `qchem` is named `"OpenFermion"`, since we initially envisioned qchem as an openfermion plugin.